crates/ark/tests/data_explorer.rs

```diff
@@ -1686,6 +1686,105 @@ fn test_update_data_filters_reapplied() {
     ]);
 }
 
+#[test]
+fn test_set_membership_filter() {
+    let _lock = r_test_lock();
+
+    // Create a data frame for testing set membership filters
+    r_task(|| {
+        harp::parse_eval_global(
+            r#"categories <- data.frame(
+                fruit = c(
+                    "apple",
+                    "banana",
+                    "orange",
+                    "grape",
+                    "kiwi",
+                    "pear",
+                    "strawberry"
+                )
+            )"#,
+        )
+        .unwrap();
+    });
+
+    let socket = open_data_explorer(String::from("categories"));
+
+    let req = DataExplorerBackendRequest::GetSchema(GetSchemaParams {
+        column_indices: vec![0],
+    });
+
+    let schema_reply = socket_rpc(&socket, req);
+    let schema = match schema_reply {
+        DataExplorerBackendReply::GetSchemaReply(schema) => schema,
+        _ => panic!("Unexpected reply: {:?}", schema_reply),
+    };
+
+    // Apply an inclusive set membership filter (include apple, banana, pear)
+    let inclusive_filter = RowFilter {
+        column_schema: schema.columns[0].clone(),
+        filter_type: RowFilterType::SetMembership,
+        filter_id: "3F984747-4667-40CB-9013-AA659AE37F1A".to_string(),
+        condition: RowFilterCondition::And,
+        is_valid: None,
+        params: Some(RowFilterParams::SetMembership(
+            amalthea::comm::data_explorer_comm::FilterSetMembership {
+                values: vec![
```

Comment:
Is it possible to represent `NA` in `values`? I have a feeling it is probably not possible, but one interesting thing about `%in%` is that `NA` is treated as a match, while with `==` it is not.

```
> NA_character_ == NA_character_
[1] NA
> NA_character_ %in% c("x", NA_character_)
[1] TRUE
```

Reply:
Currently we have a deficiency in the filter RPC protocol in that it isn't possible to distinguish between the value `NA`, so we're going to need to change that to use the same special value codes that we use to make that distinction when populating the data grid